contracts/0.4.24/nos/NodeOperatorsRegistry.sol

```diff
@@ -1,702 +1,1169 @@
+            totalSigningKeysCount =
+                SIGNING_KEYS_MAPPING_NAME.removeUnusedKeySig(_nodeOperatorId, i - 1, totalSigningKeysCount.sub(1));
+        }
+        signingKeysStats.set(TOTAL_KEYS_COUNT_OFFSET, uint64(totalSigningKeysCount));
+        emit TotalSigningKeysCountChanged(_nodeOperatorId, totalSigningKeysCount);
+
+        uint64 vettedSigningKeysCount = signingKeysStats.get(VETTED_KEYS_COUNT_OFFSET);
+        if (_fromIndex < vettedSigningKeysCount) {
+            // decreasing the staking limit so the key at _index can't be used anymore
+            signingKeysStats.set(VETTED_KEYS_COUNT_OFFSET, uint64(_fromIndex));
+            emit VettedSigningKeysCountChanged(_nodeOperatorId, _fromIndex);
+        }
+        _saveOperatorSigningKeysStats(_nodeOperatorId, signingKeysStats);
+
+        _increaseValidatorsKeysNonce();
     }
 
-    /**
-      * @notice Returns total number of node operators
-      */
-    function getNodeOperatorsCount() public view returns (uint256) {
-        return TOTAL_OPERATORS_COUNT_POSITION.getStorageUint256();
+    /// @notice Returns total number of signing keys of the node operator #`_nodeOperatorId`
+    function getTotalSigningKeyCount(uint256 _nodeOperatorId) external view returns (uint256) {
+        _onlyExistedNodeOperator(_nodeOperatorId);
+        Packed64x4.Packed memory signingKeysStats = _loadOperatorSigningKeysStats(_nodeOperatorId);
+        return signingKeysStats.get(TOTAL_KEYS_COUNT_OFFSET);
     }
 
-    /**
-     * @notice Returns a monotonically increasing counter that gets incremented when any of the following happens:
-     *   1. a node operator's key(s) is added;
-     *   2. a node operator's key(s) is removed;
-     *   3. a node operator's approved keys limit is changed.
-     *   4. a node operator was activated/deactivated. Activation or deactivation of node operator
-     *      might lead to usage of unvalidated keys in the assignNextSigningKeys method.
-     */
-    function getKeysOpIndex() public view returns (uint256) {
+    /// @notice Returns number of usable signing keys of the node operator #`_nodeOperatorId`
+    function getUnusedSigningKeyCount(uint256 _nodeOperatorId) external view returns (uint256) {
+        _onlyExistedNodeOperator(_nodeOperatorId);
+
+        Packed64x4.Packed memory signingKeysStats = _loadOperatorSigningKeysStats(_nodeOperatorId);
+        return signingKeysStats.get(TOTAL_KEYS_COUNT_OFFSET) - signingKeysStats.get(DEPOSITED_KEYS_COUNT_OFFSET);
+    }
+
+    /// @notice Returns a monotonically increasing counter that gets incremented when any of the following happens:
+    ///   1. a node operator's key(s) is added;
+    ///   2. a node operator's key(s) is removed;
+    ///   3. a node operator's vetted keys count is changed.
+    ///   4. a node operator was activated/deactivated. Activation or deactivation of node operator
+    ///      might lead to usage of unvalidated keys in the assignNextSigningKeys method.
```

Comment:
I would suggest to take a fresh description from the IStakingModule

    /// @notice Returns a counter that MUST change its value whenever the deposit data set changes.
    ///     Below is the typical list of actions that requires an update of the nonce:
    ///     1. a node operator's deposit data is added
    ///     2. a node operator's deposit data is removed
    ///     3. a node operator's ready-to-deposit data size is changed
    ///     4. a node operator was activated/deactivated
    ///     5. a node operator's deposit data is used for the deposit
    ///     Note: Depending on the StakingModule implementation above list might be extended

Reply:
and we can put it next to `getNonce` and add that it's deprecated

Reply:
fixed